example/broker.cpp

```diff
@@ -16,20 +16,77 @@
 #include <fstream>
 
 #if defined(MQTT_USE_TLS)
-boost::asio::ssl::context init_ctx(boost::program_options::variables_map const& vm)
+boost::asio::ssl::context init_ctx()
 {
-    if (vm.count("certificate") == 0 && vm.count("private_key") == 0) {
-        throw std::runtime_error("TLS requested but certificate and/or private_key not specified");
-    }
-
     boost::asio::ssl::context ctx(boost::asio::ssl::context::tlsv12);
     ctx.set_options(
         boost::asio::ssl::context::default_workarounds |
         boost::asio::ssl::context::single_dh_use);
-    ctx.use_certificate_file(vm["certificate"].as<std::string>(), boost::asio::ssl::context::pem);
-    ctx.use_private_key_file(vm["private_key"].as<std::string>(), boost::asio::ssl::context::pem);
     return ctx;
 }
+
+template<typename Server>
+void reload_ctx(Server& server, boost::asio::steady_timer& reload_timer,
+                std::string const &certificate_filename,
+                std::string const &key_filename,
+                unsigned int certificate_reload_interval,
+                char const* name, bool first_load = true)
+{
+    MQTT_LOG("mqtt_broker", info) << "Reloading certificates for server " << name;
+
+    if (certificate_reload_interval > 0) {
+        reload_timer.expires_after(std::chrono::hours(certificate_reload_interval));
+        reload_timer.async_wait([&server, &reload_timer,
+                                 certificate_filename, key_filename, certificate_reload_interval, name](boost::system::error_code const &e) {
```

Comment:
```suggestion
        reload_timer.async_wait(
            [&server, &reload_timer, certificate_filename, key_filename, certificate_reload_interval, name]
            (boost::system::error_code const& e) {
```

Reply:
The main point is `const &e` vs `const& e`. But I also suggested change line breaks. Please update it.